rudder-core/src/main/scala/com/normation/rudder/migration/XmlEntityMigration.scala

```diff
@@ -364,6 +374,76 @@ trait BatchElementMigration[T <: MigrableEntity] extends XmlFileFormatMigration
 
 
 /**
+ * The migration for eventlogs.
+ * This is the intersting part for requests
+ */
+
+trait EventLogsMigration extends BatchElementMigration[MigrationEventLog] {
+
+  override final val elementName = "EventLog"
+  override final val rowMapper = MigrationEventLogMapper
+  override final def selectAllSqlRequest(batchSize: Int) = {
+    s"select id, eventType, data from (select id, eventType, data, ((xpath('/entry//@fileFormat',data))[1]::text) as version from eventlog) as T where version='${fromVersion}' limit ${batchSize}"
+  }
+
+  override final protected def save(logs:Seq[MigrationEventLog]) : Box[Seq[MigrationEventLog]] = {
+    val UPDATE_SQL = "UPDATE EventLog set eventType = ?, data = ? where id = ?"
+
+    val ilogs = logs match {
+      case x:IndexedSeq[_] => logs
+      case seq => seq.toIndexedSeq
```

Comment:
That's not a part of the code I modified, I just moved from an other class. 
The reason may be that we need to access the sed element by index after that point, and we want to be sure that it won't fail.